sdk/identity/azure-identity/src/main/java/com/azure/identity/IdentityClient.java

```diff
@@ -173,15 +191,37 @@ public IdentityClient(IdentityClientOptions options) {
      */
     public Mono<AccessToken> authenticateWithUsernamePassword(String tenantId, String clientId, String[] scopes, String username, String password) {
         String authorityUrl = options.authorityHost().replaceAll("/+$", "") + "/" + tenantId;
-        ExecutorService executor = Executors.newSingleThreadExecutor();
         try {
-            PublicClientApplication application = PublicClientApplication.builder(clientId)
-                .authority(authorityUrl)
-                .executorService(executor)
-                .build();
+            PublicClientApplication.Builder applicationBuilder = PublicClientApplication.builder(clientId).authority(authorityUrl);
+            if (options.proxyOptions() != null) {
+                applicationBuilder.proxy(proxyOptionsToJavaNetProxy(options.proxyOptions()));
+            }
+            PublicClientApplication application = applicationBuilder.build();
             return Mono.fromFuture(application.acquireToken(UserNamePasswordParameters.builder(new HashSet<>(Arrays.asList(scopes)), username, password.toCharArray()).build()))
-                .map(ar -> new AccessToken(ar.accessToken(), OffsetDateTime.ofInstant(ar.expiresOnDate().toInstant(), ZoneOffset.UTC)))
-                .doFinally(s -> executor.shutdown());
+                .map(ar -> {
+                    currentAccount.set(ar.account());
```

Comment:
Can the identity client only authenticate one user account?  Will credential instances always have their own IdentityClient instance?  Each credential should be able to authenticate a unique user.

Reply:
Currently every credential has its own IdentityClient - I don't like this approach of a lazy initialized PublicClientApplication but it seems to get the job done. I'll play with it a little bit more.